datasets/us/finra_enforcement_actions/crawler.py

```diff
@@ -0,0 +1,90 @@
+import re
+import requests
+from typing import Generator, Dict, Tuple, Optional
+from lxml.etree import _Element
+from normality import slugify, collapse_spaces
+from zavod import Context, helpers as h
+
+BASE_URL = "https://www.finra.org"
+
+
+def parse_table(
+    table: _Element,
+) -> Generator[Dict[str, Tuple[str, Optional[str]]], None, None]:
+    headers = None
+    for row in table.findall(".//tr"):
+        if headers is None:
+            headers = []
+            for el in row.findall("./th"):
+                headers.append(slugify(el.text_content()))
+            continue
+
+        cells = []
+        for el in row.findall("./td"):
+            a = el.find(".//a")
+            if a is None:
+                cells.append((collapse_spaces(el.text_content()), None))
+            else:
+                cells.append((collapse_spaces(a.text_content()), a.get("href")))
+
+        assert len(headers) == len(cells)
+        yield {hdr: c for hdr, c in zip(headers, cells)}, row
+
+
+def crawl_item(input_dict: dict, html_element: _Element, context: Context):
+    schema = "LegalEntity"
+
+    # We try to find if it's a person or company using the icon class
+    if "user" in html_element.find(".//i").get("class"):
+        schema = "Person"
+
+    if "building" in html_element.find(".//i").get("class"):
+        schema = "Company"
+
+    name = input_dict.pop("firms-individuals")[0]
+    case_summary = input_dict.pop("case-summary")[0]
+    case_id = input_dict.pop("case-id")[0]
+    source_url = BASE_URL + case_id
```

Comment:
try make_links_absolute in crawl() then I think you won't need t construct URLs from the base https://github.com/opensanctions/opensanctions/blob/main/datasets/at/meine_abgeordneten/crawler.py#L178

Reply:
this is creating URLs like "https://www.finra.org2022077093901/", - do you mean to be using case-id[1]?

Reply:
Nice, I didn't know about this function